plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -381,113 +384,170 @@ public static void applyConfiguredNodeScaling(final ElkNode node) {
+        }
+
+        if (!allConnectedSections.isEmpty()) {
+
+            // let p1 be the start point
+            KVector p1 = sectionPoints[reverse ? sectionPoints.length - 1 : 0];
+
+            // for all bend points of this connection
+            for (int i = 1; i < sectionPoints.length; i++) {
+                // let p2 be the next bend point on this connection
+                KVector p2 = sectionPoints[reverse ? sectionPoints.length - 1 - i : i];
+
+                // for all other connections that are still on the same track as this one
+                Iterator<ElkEdgeSection> allSectIter = allConnectedSections.iterator();
+                while (allSectIter.hasNext()) {
+                    ElkEdgeSection otherSection = allSectIter.next();
+                    KVector[] otherPoints = pointsMap.get(otherSection);
+                    if (otherPoints.length <= i) {
+                        allSectIter.remove();
+                    } else {
+                        // let p3 be the next bend point of the other connection
+                        KVector p3 = new KVector(otherPoints[reverse ? otherPoints.length - 1 - i : i])
+                                .add(offsetMap.get(otherSection));
+                        if (p2.x != p3.x || p2.y != p3.y) {
+                            // the next point of this and the other connection differ
+                            double dx2 = p2.x - p1.x;
+                            double dy2 = p2.y - p1.y;
+                            double dx3 = p3.x - p1.x;
+                            double dy3 = p3.y - p1.y;
+                            if ((dx3 * dy2) == (dy3 * dx2) && Math.signum(dx2) == Math.signum(dx3)
+                                    && Math.signum(dy2) == Math.signum(dy3)) {
+
+                                // the points p1, p2, p3 form a straight line,
+                                // now check whether p2 is between p1 and p3
+                                if (Math.abs(dx2) < Math.abs(dx3) || Math.abs(dy2) < Math.abs(dy3)) {
+                                    junctionPoints.add(p2);
+                                }
+
+                            } else if (i > 1) {
+                                // p2 and p3 have diverged, so the last common point is p1
+                                junctionPoints.add(p1);
+                            }
+
+                            // do not consider the other connection in the next iterations
+                            allSectIter.remove();
+                        }
+                    }
+                }
+                // for the next iteration p2 is taken as reference point
+                p1 = p2;
+            }
+        }
+
+        return junctionPoints;
+    }
+
+    /**
+     * Get the edge section points as an array of vectors.
+     * 
+     * @param section an edge section
+     * @return an array with all edge section points
+     */
+    private static KVector[] getPoints(final ElkEdgeSection section) {
```

Comment:
Duplicates the code of `ElkUtil#createVectorChain(ElkEdgeSection)`, which returns a `KVectorChain`. Yet, since the chain extends `LinkedList` it's not the best choice for the code above.

Reply:
Even worse: It duplicates `ElkUtil#getPoints(ElkEdge)` which it should replace. I wrote the fix in the 0.2.2 codebase and missed the moved method. I will remove the old `getPoints(ElkEdge)`